runtime/src/chpl-gpu.c

```diff
@@ -681,6 +698,13 @@ void chpl_gpu_arg_reduce(void* _cfg, void* arg, size_t elem_size,
   }
 }
 
+void chpl_gpu_arg_host_register(void* _cfg, void* arg, size_t size) {
+  kernel_cfg* cfg = (kernel_cfg*)_cfg;
+  cfg_add_direct_param(cfg, arg);
+  chpl_gpu_impl_host_register(*((void**)arg), sizeof(char)*8);
```

Comment:
I was curious about the AMD difference. I am totally fine this going in as-is. But in case it is helpful, I'd imagine this `impl` function could return the device pointer, which would be the same as `arg` for nvidia and different for amd. Then, that pointer will have to be passed to `cfg_add_direct_param` instead after the call to `chpl_gpu_impl_host_register`. Does that sound right? Do you think there's more to it?

Reply:
I think that's exactly right.

I was thinking I'd have to add some new structure to the `cfg` to say how to "remap" the arguments, but you're right I can just update it directly by changing what gets passed to `cfg_add_direct_param`.

Reply:
I see. I can't think of any need for retaining that mapping at the moment. Maybe there's a function in hip/cuda that can do that for us if we need it?

One thing that could potentially need that is unregistering the memory, but it looks like we pass the initial host address for that as well (and that makes sense).